### PR TITLE
1532-group-agreements

### PR DESCRIPTION
**Pull request by tibetsprague**
Group Settings page
Show in group welcome modal and make people accept all agreements before viewing group

Fixes https://github.com/Hylozoic/hylo-evo/issues/1532

### CODE REVIEW

**Review by thomasgwatson — APPROVED** [review]
lgtm

**Comment by thomasgwatson** [comment]
Well, the tests are failing for this. Probably just need to update the test database schema?

File: src/routes/UserSettings/UserSettings.store.js [hunk 1]
```diff
@@ -165,13 +165,15 @@ export function updateMembershipSettings (groupId, settings) {
       `,
       variables: {
         data: {
+          acceptAgreements,
```
[the remark this answers is not in the document]
Comment (tibetsprague) [reply src/routes/UserSettings/UserSettings.store.js:168]: no, passing this in checks off all the current agreememts as accepted

File: src/store/reducers/ormReducer/index.js [hunk 2]
```diff
@@ -500,6 +500,11 @@ export default function ormReducer (state = orm.getEmptyState(), action) {
         group = Group.withId(meta.id)
         clearCacheFor(Group, meta.id)
       }
+
+      if (payload.data.updateGroupSettings && (payload.data.updateGroupSettings.agreements)) {
+        group = Group.withId(meta.id)
+        clearCacheFor(Group, meta.id)
```
[the remark this answers is not in the document]
Comment (tibetsprague) [reply src/store/reducers/ormReducer/index.js:506]: it makes sure that everywhere the group is accessed client side it has the latest data

**Pull request merged**